solr/core/src/java/org/apache/solr/handler/admin/api/RestoreCoreAPI.java

```diff
@@ -146,6 +147,10 @@ private void doRestore(String coreName, RestoreCoreRequestBody requestBody) thro
           .getZkController()
           .getShardTerms(cd.getCollectionName(), cd.getShardId())
           .ensureHighestTermsAreNotZero();
+
+      // transitions state of update log to ACTIVE
+      UpdateLog updateLog = core.getUpdateHandler().getUpdateLog();
+      updateLog.applyBufferedUpdates();
```

Comment:
I'd add a null check on this updateLog just in case there is none.  In practice it's always there for SolrCloud but there are lots of null checks elsewhere for it and I like to think we could run SolrCloud without insisting on it (albeit expert).